src/components/AddBikeDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { makeStyles, Theme, createStyles, FormControl, InputLabel, MenuItem, Select } from '@material-ui/core';
+import { useSnackbar } from 'notistack';
+import { Station } from '../models/station';
+import { addBike } from '../api/bikes/addBike';
+import { Bike } from '../models/bike';
+import { getAllStations } from '../api/stations/getStations';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    content: {
+      flexGrow: 1,
+      padding: theme.spacing(3),      
+      display: "flex",
+      width: "100%",
+      justify: "center", 
+      alignItems: "center",      
+    },
+    table: {
+      minWidth: 650,
+    },
+    addButton: {
+      margin: theme.spacing(2),
+    },
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+      },
+  })
+);
+
+export interface AddBikeDialogProps {
+  setBikes: (value: React.SetStateAction<Bike[]>) => void;  
+}
+
+const AddBikeDialog = (props: AddBikeDialogProps) => {
+  const classes = useStyles()
+  const [open, setOpen] = React.useState<boolean>(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleCancel = () => {
+    setOpen(false);
+  };
+  
+  const [stations, setStations] = useState<Station[]>([]);  
+  const [stationId, setStationId] = useState<string>('');
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setStationId(event.target.value as string);
+  };
+
+  const handleSubmit = () => {
+    const bike = addBike(stationId);
+    bike.then(response => {
+      if(response.isError)
+      {
+        enqueueSnackbar("Could not add bike", { variant: "error" });
+      }
+      else if(response.data)
+      {
+        props.setBikes(prev => 
+          response.data ? 
+          prev = [...prev, response.data] : prev
```

Comment:
response.data will always return true, because you are checking it in elseif above

Reply:
`prev = [...prev, response.data]` is wrong. Make it `[...prev, response.data]`.

Reply:
look at my suggestion below